pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -327,10 +328,16 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 			allVolumesAdded = false
 			continue
 		}
+		uniqueVolumeName, err := dswp.desiredStateOfWorld.GetUniqueVolumeName(uniquePodName, volumeSpec)
+		if err != nil || !dswp.desiredStateOfWorld.PodExistsInVolume(uniquePodName, uniqueVolumeName) {
```

Comment:
Can you add a lot of comments explaining why we need to do this? It's very subtle and I think we should document it here so we don't lose history.

Reply:
Also, is this really solving the problem? I thought the problematic call is from `createVolumeSpec`, which is called before this?

Reply:
Please see the new test which fails without this PR.

Reply:
@msau42 
I have added comment above introduction of allVolumesAddedPreviously variable.
Please take a look.